MSAL/src-internal/ios/MSALLegacySharedAccount.m

```diff
@@ -0,0 +1,211 @@
+        MSID_LOG_WITH_CTX_PII(MSIDLogLevelInfo, nil, @"Created sign in status dictionary %@", MSID_PII_LOG_MASKABLE(_signinStatusDictionary));
+    }
+    
+    return self;
+}
+
+- (instancetype)initWithMSALAccount:(id<MSALAccount>)account
+                      accountClaims:(NSDictionary *)claims
+                    applicationName:(NSString *)appName
+                     accountVersion:(MSALLegacySharedAccountVersion)accountVersion
+                              error:(NSError **)error
+{
+    if (accountVersion == MSALLegacySharedAccountVersionV1)
+    {
+        return nil;
+    }
+    
+    if (!account)
+    {
+        if (error)
+        {
+            *error = MSIDCreateError(MSIDErrorDomain, MSIDErrorInternal, @"Unexpected parameter - no account", nil, nil, nil, nil, nil);
+        }
+        
+        return nil;
+    }
+    
+    NSString *appBundleId = [[NSBundle mainBundle] bundleIdentifier];
+    
+    NSMutableDictionary *jsonDictionary = [NSMutableDictionary new];
+    jsonDictionary[@"id"] = [[NSUUID UUID] UUIDString];
+    jsonDictionary[@"environment"] = @"PROD";
+    
+    if (accountVersion == MSALLegacySharedAccountVersionV3)
+    {
+        jsonDictionary[@"originAppId"] = appBundleId;
+    }
+    
+    jsonDictionary[@"signInStatus"] = @{appBundleId : @"SignedIn"};
+    jsonDictionary[@"username"] = account.username;
+    jsonDictionary[@"additionalProperties"] = @{@"createdBy": appName};
+    [jsonDictionary addEntriesFromDictionary:[self claimsFromMSALAccount:account claims:claims]];
+    return [self initWithJSONDictionary:jsonDictionary error:error];
+}
+
+#pragma mark - Match
+
+- (BOOL)matchesParameters:(MSALAccountEnumerationParameters *)parameters
+{
+    if (parameters.needsAssociatedRefreshToken)
+    {
+        NSString *appIdentifier = [[NSBundle mainBundle] bundleIdentifier];
+        NSString *signinStatus = _signinStatusDictionary[appIdentifier];
+        
+        if (![signinStatus isEqualToString:@"SignedIn"])
+        {
+            return NO;
+        }
+        
+        return YES;
+    }
+    
+    return YES;
+}
+
+#pragma mark - Update
+
+- (BOOL)updateAccountWithMSALAccount:(id<MSALAccount>)account
+                     applicationName:(NSString *)appName
+                           operation:(MSALLegacySharedAccountWriteOperation)operation
+                      accountVersion:(MSALLegacySharedAccountVersion)accountVersion
+                               error:(NSError **)error
+{
+    if (accountVersion == MSALLegacySharedAccountVersionV1)
+    {
+        return YES;
+    }
+    
+    NSMutableDictionary *oldDictionary = [self.jsonDictionary mutableCopy];
+    NSString *appIdentifier = [[NSBundle mainBundle] bundleIdentifier];
+    
+    if (appIdentifier)
+    {
+        NSMutableDictionary *signinDictionary = [NSMutableDictionary new];
+        [signinDictionary addEntriesFromDictionary:_signinStatusDictionary];
```

Comment:
nit: Can we either use either dot notation or instance variable directly everywhere to be consistent.

Reply:
+1, apple recommends using properties vs instance var.

Reply:
Addressed